docs-src/dev-guide/major-components.md

```diff
@@ -0,0 +1,263 @@
+
+### Temporal Client
+
+A Temporal Client acts as a bridge of communication between the systems outside of Temporal and your application.
+The Client performs key functions that facilitate the execution of, management of, and communication with Workflows.
+
+The following code is an example using the Go SDK.
+It showcases how to initialize a Temporal Client, create a connection to a local Temporal Cluster, and start a Workflow Execution:
+
+:::caution Do not copy and use code
+
+The following code is for example purposes only.
+For tested code samples and best practices, use your preferred language SDK's developer's guide.
+
+:::
+
+```go
+package main
+
+import (
+	"context"
+
+	"go.temporal.io/sdk/client"
+)
+
+func main() {
+  // Temporal Client setup code
+	c, err := client.NewClient(client.Options{})
+	if err != nil {
+		log.Fatalln("Unable to create client", err)
+	}
+	defer c.Close()
+  // Prepare Workflow option and parameters
+	workflowOptions := client.StartWorkflowOptions{
+		ID:        "loan-application-1",
+		TaskQueue: "loan-application-task-queue",
+	}
+	applicantDetails := ApplicantDetails{
+		// ...
+	}
+  // Start the Workflow
+	workflowRun, err := c.ExecuteWorkflow(context.Background(), workflowOptions, "loan-application-workflow", applicantDetails)
+	if err != nil {
+		// ...
+	}
+	// ...
+}
+```
+
+Developers can then use the Client as the main entry point for interacting with the application through Temporal.
+From here, developers can now start or Signal Workflows, query the Workflow state etc.
+We can see in the example above how the developer has used the provided start method to start the Workflow upon the running of the function.
+
+### APIs to Develop Workflows
+
+Workflows are defined as code as either a function or an object method, depending on the language.
+
+For example, the following is a valid Temporal Workflow in Go:
+
+:::caution Do not copy and use code
+
+The following code is for example purposes only.
+For tested code samples and best practices, use your preferred language SDK's developer's guide.
+
+:::
+
+```go
+func LoanApplication(ctx context.Context) (error) {
+	// ...
+	return nil
+}
+```
+
+The Workflow code uses Temporal SDK APIs to orchestrate the steps of the application.
+
+:::caution Do not copy and use code
+
+The following code is for example purposes only.
+For tested code samples and best practices, use your preferred language SDK's developer's guide.
+
+:::
+
+```go
+func LoanApplication(ctx workflow.Context, input *LoanApplicationWorkflowInput) (*LoanApplicationWorkflowResult, error) {
+	// ...
+	var result activities.CreditCheckResult
+	f := workflow.ExecuteActivity(ctx, a.CreditCheck, CreditCheckInput(*input))
+	err := f.Get(ctx, &result)
+	// ...
+	// Return the results
+	return &loanApplicationResults, nil
+}
+```
+
+A Workflow executes Activities (other functions), handles and sends messages (Queries, Signals, Updates), and interacts with other Workflows.
```

Comment:
"other functions" is a bit vague to me..how about other functions that interact with external systems, functions that work with non-deterministic code etc.